### PR TITLE
Added BlockDetonateEvent to be fired when a TNT gets detonated

### PR DESCRIPTION
**Pull request by nitnelave**
This event allows to detect and cancel a detonation event (when the TNT block is replaced by a blinking entity).

### CODE REVIEW

**Comment by nitnelave** [comment]
CraftBukkit PR : https://github.com/Bukkit/CraftBukkit/pull/917

**Comment by nitnelave** [comment]
And later we could add detonation reasons, like redstone, explosion, or fire.

**Comment by lol768** [comment]
Don't we already have EntityExplodeEvent? You could just do a getEntityType() and check if it is PRIMED_TNT.

EDIT: I see, this event is before the conversion into an entity.

**Comment by nitnelave** [comment]
I have to check, but I think entityChangeBlockEvent now covers this scenario.

**Comment by riking** [comment]
You need a longer description - convince us. Why is this implementation better than past ones? Why should Bukkit accept this TNT implementation and not others? Do you include ENDER_CRYSTALs?

> And later we could add detonation reasons, like redstone, explosion, or fire.

@nitnelave Once an API is submitted, you CANNOT change it without being "breaking". Finalize before submitting.

**Comment by authorblues** [comment]
I stand by what I have said in the past: this is a bad solution to add a **third** event related to explosions. This is precisely what ExplosionPrimeEvent advertised, and exactly what it was originally intended for. Just because its implementation is patently wrong doesn't mean that we need yet another event.

But let's actually fix this bug, please.

File: src/main/java/org/bukkit/event/block/BlockDetonateEvent.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package org.bukkit.event.block;
+
+import org.bukkit.block.Block;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.HandlerList;
+
+/**
+ * Called when a TNT block is detonated (starts blinking).
```
Comment (natemort) [inline src/main/java/org/bukkit/event/block/BlockDetonateEvent.java]: When TNT starts blinking, that's when it has been ignited, not detonated.  Detonation would be the explosion.

Comment (nitnelave) [inline src/main/java/org/bukkit/event/block/BlockDetonateEvent.java]: Detonated can also mean the ignition part, but I agree Ignited is clearer than detonated. I have to work on that PR again.

**Pull request merged**